contracts/src/v0.8/dev/ArbitrumCrossDomainGovernor.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./interfaces/DelegateForwarderInterface.sol";
+import "./vendor/arb-bridge-eth/v0.8.0-custom/contracts/libraries/AddressAliasHelper.sol";
+import "./vendor/openzeppelin-solidity/v4.3.1/contracts/utils/Address.sol";
+import "./ArbitrumCrossDomainForwarder.sol";
+
+/**
+ * @title ArbitrumCrossDomainGovernor - L1 xDomain account representation (with delegatecall support) for Arbitrum
+ * @notice L2 Contract which receives messages from a specific L1 address and transparently forwards them to the destination.
+ * @dev Any other L2 contract which uses this contract's address as a privileged position,
+ *   can be considered to be simultaneously owned by the `l1Owner` and L2 `owner`
+ */
+contract ArbitrumCrossDomainGovernor is DelegateForwarderInterface, ArbitrumCrossDomainForwarder {
+  /**
+   * @notice creates a new Arbitrum xDomain Forwarder contract
+   * @param l1OwnerAddr the L1 owner address that will be allowed to call the forward fn
+   * @dev Empty constructor required due to inheriting from abstract contract CrossDomainForwarder
+   */
+  constructor(address l1OwnerAddr) ArbitrumCrossDomainForwarder(l1OwnerAddr) {}
+
+  /**
+   * @notice versions:
+   *
+   * - ArbitrumCrossDomainGovernor 1.0.0: initial release
+   *
+   * @inheritdoc TypeAndVersionInterface
+   */
+  function typeAndVersion() external pure virtual override returns (string memory) {
+    return "ArbitrumCrossDomainGovernor 1.0.0";
+  }
+
+  /**
+   * @dev forwarded only if L2 Messenger calls with `msg.sender` being the L1 owner address, or called by the L2 owner
+   * @inheritdoc ForwarderInterface
+   */
+  function forward(address target, bytes memory data) external override onlyCrossDomainOwners {
+    Address.functionCall(target, data, "Governor call reverted");
+  }
+
+  /**
+   * @dev forwarded only if L2 Messenger calls with `msg.sender` being the L1 owner address, or called by the L2 owner
+   * @inheritdoc DelegateForwarderInterface
+   */
+  function forwardDelegate(address target, bytes memory data) external override onlyCrossDomainOwners {
+    Address.functionDelegateCall(target, data, "Governor delegatecall reverted");
+  }
+
+  /**
+   * @notice The call MUST come from either the L1 owner (via cross-chain message) or the L2 owner. Reverts otherwise.
+   */
+  modifier onlyCrossDomainOwners() {
```

Comment:
Consider renaming this modifier as it's not the x-domain owner who has access, but also the local owner.

Maybe simple `onlyOwners`?

Reply:
I think `onlyOwners` is too similar to `onlyOwner` and will be more confusing. The cross domain owners here refers to the following set of owners:
- L1 owner
- L2 owner

Reply:
Yes, not a great suggestion I agree, but `onlyCrossDomainOwners` really doesn't fit as it includes L2 owner which is not x-domain to the contract in context.

Reply:
`onlySelectedOwners`?

Reply:
How about:
- `onlyCrossDomainOrLocalOwner`
- `onlyGovernors`
- `onlyGovernorOwners`

Reply:
Renamed to `onlyLocalOrCrossDomainOwner`

Reply:
That sounds explicit and nice, albeit a little verbose. I like it more than the original in any case. Thx!